### DISCUSSION TITLE
Debouncing of x-model in case of multiple checkboxes

### DISCUSSION THREAD

**Discussion by iljamm** [alpinejs/alpine#3747] [3. Feature ideas]
Don't know if this one is a bug, I am sure there are reasons for how things work as they do, so I made it a feature request and not a bug report. 

I have this code:

```
<div x-data="boxes">
  <template x-for="i in 10">
    <input type="checkbox" :value="i" x-model.debounce.500ms="_items">
  </template>
  <div x-text="_items"></div>
</div>

<script>
  document.addEventListener("alpine:init", function() {
    Alpine.data("boxes", function() {
      return {
        _items: [],
      };
    });
  });
</script>
```

[https://codepen.io/tirins007/pen/yLGaKOa](https://codepen.io/tirins007/pen/yLGaKOa)

As you can see I added "debounce" modifier to each checkbox's x-model. My expectation was that the component property to which these checkboxes are linked (_item in my case) would be debounced, and therefore the checkbox values array would be saved into it once after I stop clicking checkboxes in a quick succession. I would like to be able to quickly check a number of checkboxes and after 500ms delay values of all checked checkboxes would be saved into _items property.

Presently debouncing simply deactivates checkbox clicks.

I have come up with a work around, and this codepen shows how I would like to see the x-model debounce to work:

[https://codepen.io/tirins007/pen/QWzKmpy](https://codepen.io/tirins007/pen/QWzKmpy)

**Comment by ekwoka** [comment]
hmm, It definitely debounces the event at the element.

I'm not sure there is a non-spaghetti way to handle this in the library, and not just do it with getter/setter pair with a debounced assignment function...

**Reply by iljamm** [reply]
@ekwoka could you possibly expand on the "getter/setter pair" approach. Would you still be using x-model to achieve the effect?

**Reply by ekwoka** [reply]
Actually, I just realized that with how it updates checkbox arrays, a getter setter pair won't actually work.

You'd need to manually do `:checked` and `@change.debounce`

🤔
